src/passes/LegalizeJSInterface.cpp

```diff
@@ -241,50 +242,73 @@ struct LegalizeJSInterface : public Pass {
     return func->name;
   }
 
-  void ensureTempRet0(Module* module) {
-    if (!module->getGlobalOrNull(TEMP_RET_0)) {
-      module->addGlobal(Builder::makeGlobal(
-        TEMP_RET_0,
-        i32,
-        LiteralUtils::makeZero(i32, *module),
-        Builder::Mutable
-      ));
-      needTempRet0Helpers = true;
-    }
-  }
 
   void addTempRet0Helpers(Module* module) {
-    // We should also let JS access the tempRet0 global, which
-    // is necessary to send/receive 64-bit return values.
+    bool getterExists = module->getExportOrNull(GET_TEMP_RET_0) != nullptr;
+    bool setterExists = module->getExportOrNull(SET_TEMP_RET_0) != nullptr;
+
+    if (module->getGlobalOrNull(TEMP_RET_0)) {
+      // The global already exists.   Ensure we also have the getter and setter.
+      if (getterExists) {
```

Comment:
Should these be `if (!getterExists) {`?
As written this should always fail if the getter or setter already exists... and it looks like it does, so we're not missing test coverage.